src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php

```diff
@@ -192,11 +193,22 @@ public function getTypes($class, $property, array $context = [])
                         case self::$useDeprecatedConstants ? DBALType::TARRAY : Types::ARRAY:
                         // no break
                         case 'json_array':
+                            // return null if $enumType is set, because we can't determine if collectionKeyType is string or int
+                            if ($enumType) {
+                                return null;
+                            }
+
                             return [new Type(Type::BUILTIN_TYPE_ARRAY, $nullable, null, true)];
 
                         case self::$useDeprecatedConstants ? DBALType::SIMPLE_ARRAY : Types::SIMPLE_ARRAY:
-                            return [new Type(Type::BUILTIN_TYPE_ARRAY, $nullable, null, true, new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_STRING))];
+                            return [new Type(Type::BUILTIN_TYPE_ARRAY, $nullable, null, true, new Type(Type::BUILTIN_TYPE_INT), $enumType ?? new Type(Type::BUILTIN_TYPE_STRING))];
+                    }
+                case Type::BUILTIN_TYPE_INT:
```

Comment:
I may have merged to fast here.
@Sajito is the missing break before this line on purpose? Can you please send a PR that adds either the "break" or the "no break" comment, with a test case ideally?

Reply:
I have sent a new PR here: https://github.com/symfony/symfony/pull/46713.
I didn't know a "no break" comment would be required here, as I took these lines as a reference:
https://github.com/symfony/symfony/blob/7fc7acb1e4448762f438af03cadb2ce54e3918e0/src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php#L169-L175

Hope it's fine with the new PR.